keras/backend/tensorflow_backend.py

```diff
@@ -399,7 +399,7 @@ def gradients(loss, variables):
 # CONTROL FLOW
 
 def rnn(step_function, inputs, initial_states,
-        go_backwards=False, mask=None):
+        go_backwards=False, mask=None, constants=None):
```

Comment:
I attempted that at first, but it seemed to be slower and required more memory (with theano) than passing these arguments as non_sequences (constants). 
I did an experiment with a sentiment analysis dataset (10620 sequences of length 200). Time and memory per epoch were (on a Titan X GPU):
17s, 336MiB normal LSTM (original implementation)
18s, 412MiB dropout LSTM with constants (non_sequences)
21s, 638MiB dropout LSTM passing states